File: src/de/saschahlusiak/freebloks/view/opengl/ViewElement.java
package de.saschahlusiak.freebloks.view.opengl;

import android.graphics.PointF;

public class ViewElement {
	ViewModel model;
	
	ViewElement(ViewModel model) {
		this.model = model;
	}
	
	boolean handlePointerDown(PointF m) { return false; }
	boolean handlePointerMove(PointF m) { return false; }
	boolean handlePointerUp(PointF m)   { return false; }
}

File: src/main/java/de/saschahlusiak/freebloks/view/scene/ViewElement.kt
package de.saschahlusiak.freebloks.view.scene

import android.graphics.PointF

interface ViewElement {
    fun handlePointerDown(m: PointF): Boolean
    fun handlePointerMove(m: PointF): Boolean
    fun handlePointerUp(m: PointF): Boolean

    fun execute(elapsed: Float): Boolean
}